workerpool/pool.go

```diff
@@ -151,22 +154,52 @@ func newWorker(id uint, conn *connection.Connection, wg *sync.WaitGroup, log *za
 		done:       make(chan bool),
 		waitGroup:  wg,
 		log:        log,
+		gateway: &http.Client{
+			Timeout: 2 * time.Second,
```

Comment:
hmm, looks like it should be fine to just set Timeout https://golang.org/pkg/net/http/#Client?